eth/api.go

```diff
@@ -260,15 +260,20 @@ func (api *DebugAPI) DumpBlock(blockNr rpc.BlockNumber) (state.Dump, error) {
 		OnlyWithAddresses: true,
 		Max:               AccountRangeMaxResults, // Sanity limit over RPC
 	}
-	if blockNr == rpc.PendingBlockNumber {
-		// If we're dumping the pending state, we need to request
-		// both the pending block as well as the pending state from
-		// the miner and operate on those
-		_, stateDb := api.eth.miner.Pending()
-		return stateDb.RawDump(opts), nil
-	}
 	var block *types.Block
-	if blockNr == rpc.LatestBlockNumber {
+	if blockNr == rpc.PendingBlockNumber {
+		if api.eth.miner == nil {
+			// arbitrum: in case of ArbEthereum, miner is not available here
+			// use current block instead of pending
+			block = api.eth.blockchain.CurrentBlock()
+		} else {
+			// If we're dumping the pending state, we need to request
+			// both the pending block as well as the pending state from
+			// the miner and operate on those
+			_, stateDb := api.eth.miner.Pending()
+			return stateDb.RawDump(opts), nil
```

Comment:
Instead of this, could we do the same thing you did below to simply change the blockNr?